### ISSUE TITLE
Error when trying to set timer in home assistant blueprint

### ISSUE THREAD

**Issue by ayeganov** [cisco/mindmeld#274]
# Set Timer Bug

### Steps To Repro
1. Go through the documentation to setup the home assistant blueprint
2. Make sure to start `duckling` and `elastic search` using docker:
    1. `docker run -p 0.0.0.0:7151:7151 mindmeldworkbench/duckling:master -ti -d`
    2. `docker run -d -p 9200:9200 -p 9300:9300 -e "discovery.type=single-node" docker.elastic.co/elasticsearch/elasticsearch:7.8.0`
3. Run the conversation command: `python -m home_assistant converse`
4. Type in: `set timer for 3 minutes`

### Expected Behavior
A message indicating a success and informing the user that the timer has been set

### Actual Behavior
A stack trace is produced:

```
  File "/home/ainindza/.pyenv/versions/3.6.12/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/ainindza/.pyenv/versions/3.6.12/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/ainindza/code/ayeganov/home_assistant/home_assistant/__main__.py", line 8, in <module>
    app.cli()
  File "/home/ainindza/.pyenv/versions/dude_assistant/lib/python3.6/site-packages/mindmeld/app.py", line 241, in cli
    app_cli(obj={"app": self})
  File "/home/ainindza/.pyenv/versions/dude_assistant/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/ainindza/.pyenv/versions/dude_assistant/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/ainindza/.pyenv/versions/dude_assistant/lib/python3.6/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ainindza/.pyenv/versions/dude_assistant/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ainindza/.pyenv/versions/dude_assistant/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/ainindza/.pyenv/versions/dude_assistant/lib/python3.6/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/ainindza/.pyenv/versions/dude_assistant/lib/python3.6/site-packages/mindmeld/cli.py", line 347, in converse
    responses = convo.say(message)
  File "/home/ainindza/.pyenv/versions/dude_assistant/lib/python3.6/site-packages/mindmeld/components/dialogue.py", line 1334, in say
    response = self.process(text, params=params)
  File "/home/ainindza/.pyenv/versions/dude_assistant/lib/python3.6/site-packages/mindmeld/components/dialogue.py", line 1402, in process
    verbose=self.verbose,
  File "/home/ainindza/.pyenv/versions/dude_assistant/lib/python3.6/site-packages/mindmeld/app_manager.py", line 215, in parse
    request, response, **dm_params
  File "/home/ainindza/.pyenv/versions/dude_assistant/lib/python3.6/site-packages/mindmeld/components/dialogue.py", line 375, in apply_handler
    request, responder, target_dialogue_state=target_dialogue_state
  File "/home/ainindza/.pyenv/versions/dude_assistant/lib/python3.6/site-packages/mindmeld/components/dialogue.py", line 391, in _apply_handler_sync
    request, responder, target_dialogue_state=target_dialogue_state
  File "/home/ainindza/.pyenv/versions/dude_assistant/lib/python3.6/site-packages/mindmeld/components/dialogue.py", line 424, in _attempt_handler_sync
    res = handler(request, responder)
  File "/home/ainindza/code/ayeganov/home_assistant/home_assistant/times_and_dates.py", line 130, in start_timer
    selected_duration = _get_duration(request)
  File "/home/ainindza/code/ayeganov/home_assistant/home_assistant/times_and_dates.py", line 241, in _get_duration
    if len(duration_entity_candidates) == 0 else duration_entity_candidates[0]
TypeError: object of type 'NoneType' has no len()
```

### Initial Analysis
In short - `times_and_dates.py` uses `get_candidates_for_text` from `mindmeld.ser` module, which internally delegates to duckling. The function `_get_duration` never expects `get_candidates_for_text` to return `None`, which in this case it does - thus causing the above stack trace. As it stands now - the culprit seems to be duckling.

**Comment by vijay120** [comment]
Can you try this after stopping all existing docker contains:

```
docker run -ti -d -p 0.0.0.0:7151:7151 mindmeldworkbench/duckling:master
docker run -ti -d -p 0.0.0.0:9200:9200 -p 0.0.0.0:9300:9300 -e "discovery.type=single-node" docker.elastic.co/elasticsearch/elasticsearch:7.8.0
python -m home_assistant converse
```

And then type the query?

The above setup works for me.

**Comment by ayeganov** [comment]
@vijay120 Unfortunately this didn't work. I setup the blueprint in the virtual environment. Can we run the duckling in a way to see its log?

**Comment by vijay120** [comment]
Issue fixed by updating mindmeld version: `pip install mindmeld==4.3.4rc3`
